### ISSUE TITLE
Don't equal to CE Loss when gamma=0 and alpha=1

### ISSUE THREAD

**Issue by CherryChou98** [HeyLynne/FocalLoss_for_multiclass#5]
My case is a multi-class classification problem,
and the input dimension is:
logits: (batch_size, num_classes)
labels: (batch_size)

So I modify the codes into:
```
class FocalLoss(nn.Module):
    def __init__(self, gamma = 2, alpha = 1, size_average = True):
        super(FocalLoss, self).__init__()
        self.gamma = gamma
        self.alpha = alpha
        self.size_average = size_average
        self.elipson = 0.000001
    
    def forward(self, logits, labels):
        assert(logits.size(0) == labels.size(0))
        batch_size = logits.size(0)
        labels_length = logits.size(1)

        # transpose labels into labels onehot
        new_label = labels.unsqueeze(1)
        label_onehot = torch.zeros([batch_size, labels_length]).scatter_(1, new_label, 1)

        # calculate log
        log_p = F.log_softmax(logits, dim=1)
        pt = label_onehot * F.softmax(logits, dim=1)
        sub_pt = 1 - pt
        fl = -self.alpha * (sub_pt)**self.gamma * log_p 
        if self.size_average:
            return fl.mean()
        else:
            return fl.sum()
```

Then I test the codes with the following inputs:
```
if __name__ == "__main__":
    logits = torch.tensor([[-0.0111, -0.2735,  0.2379],
        [ 0.3806, -0.5423,  0.1786],
        [ 0.2710, -0.4325,  0.0784]])
    
    labels = torch.LongTensor([2, 0, 1])
    
    fl = FocalLoss(gamma = 0, alpha = 1.0)
    print(fl(logits, labels))
    loss_fct = CrossEntropyLoss()
    seq_loss = loss_fct(F.softmax(logits, dim=1), labels)
    print(seq_loss)
```

This time when I set `gamma = 0, alpha = 1.0`, 
the computed focal_loss is `tensor(1.1426)`,
and the CrossEntropyLoss is `tensor(1.0571)`.
These two values should be equal when `gamma = 0, alpha = 1.0`,
could you find where is wrong?
